crates/nu-command/src/filesystem/cp.rs

```diff
@@ -355,8 +355,18 @@ fn copy_file(src: PathBuf, dst: PathBuf, span: Span) -> Value {
             let message = format!("copy file {src:?} failed: {e}");
 
             use std::io::ErrorKind;
+            let dst_info = std::fs::metadata(&dst);
             let shell_error = match e.kind() {
-                ErrorKind::NotFound => ShellError::FileNotFoundCustom(message, span),
+                ErrorKind::NotFound => {
+                    if dst_info.is_ok() {
+                        ShellError::FileNotFoundCustom(message, span)
+                    } else {
+                        ShellError::FileNotFoundCustom(
+                            format!("copy file {dst:?} failed: {e}"),
+                            span,
+                        )
```

Comment:
`ErrorKind::NotFound` is not the only possible case where the target matters (e.g. `PermissionDenied`)
Maybe worth lifting the `message` generation instead of once using the variable declared before or using the local format string.